#! /usr/bin/env python

PACKAGE='controllers'

import roslib;
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       	Name		Type      	Level	Description		Default 	Min	Max 	Values
gen.add( 	"k_p",		double_t, 	0, 	"Proportional gain", 	2.0, 		0, 	10) 		# Slow: 2.0,	Fast: 4.0
gen.add( 	"k_i",		double_t, 	0, 	"Integral gain", 	0.1, 		0, 	10) 		# Slow: 0.1,	Fast: 0.2
gen.add( 	"k_d",		double_t, 	0, 	"Derivative gain", 	0.6, 		0, 	10) 		# Slow: 0.6,	Fast: 1.0

gen.add( 	"alpha_p",	double_t, 	0, 	"Proportional FOU", 	0.5, 		0, 	1) 
gen.add( 	"alpha_i",	double_t, 	0, 	"Integral FOU", 	0.5, 		0, 	1)
gen.add( 	"alpha_d", 	double_t, 	0, 	"Derivative FOU", 	0.5, 		0, 	1)

exit(gen.generate(PACKAGE, "controllers", "setSIT2FLC"))
